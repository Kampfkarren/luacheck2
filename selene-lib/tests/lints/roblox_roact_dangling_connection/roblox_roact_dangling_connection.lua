local React = a

a:connect()

local function c()
    a:Connect()
    a:connect()
    a:ConnectParallel()
    a:Once()

    a.Connect()
    a.connect()
    a.ConnectParallel()
    a.Once()

    a(b:Connect())
    a(function() end, b:Connect())

    useEffect(function()
        a:connect()
        a(b:Connect())
        local b = a:connect()
    end)

    React.useEffect(function()
        a:connect()
        a(b:Connect())
        local b = a:connect()
    end)
end
