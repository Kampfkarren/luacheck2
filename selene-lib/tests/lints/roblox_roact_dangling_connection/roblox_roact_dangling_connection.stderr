error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
  ┌─ roblox_roact_dangling_connection.lua:7:5
  │
7 │     a:Connect()
  │     ^^^^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
  ┌─ roblox_roact_dangling_connection.lua:8:5
  │
8 │     a:connect()
  │     ^^^^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
  ┌─ roblox_roact_dangling_connection.lua:9:5
  │
9 │     a:ConnectParallel()
  │     ^^^^^^^^^^^^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
   ┌─ roblox_roact_dangling_connection.lua:10:5
   │
10 │     a:Once()
   │     ^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
   ┌─ roblox_roact_dangling_connection.lua:12:5
   │
12 │     a.Connect()
   │     ^^^^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
   ┌─ roblox_roact_dangling_connection.lua:13:5
   │
13 │     a.connect()
   │     ^^^^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
   ┌─ roblox_roact_dangling_connection.lua:14:5
   │
14 │     a.ConnectParallel()
   │     ^^^^^^^^^^^^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection where appropriate
   ┌─ roblox_roact_dangling_connection.lua:15:5
   │
15 │     a.Once()
   │     ^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection in the useEffect cleanup function
   ┌─ roblox_roact_dangling_connection.lua:22:9
   │
22 │         a:connect()
   │         ^^^^^^^^^^^

error[roblox_roact_dangling_connection]: disconnect the connection in the useEffect cleanup function
   ┌─ roblox_roact_dangling_connection.lua:28:9
   │
28 │         a:connect()
   │         ^^^^^^^^^^^

